pandas/core/indexes/base.py

```diff
@@ -3059,11 +3060,14 @@ def _get_nearest_indexer(self, target: "Index", limit, tolerance) -> np.ndarray:
             right_indexer,
         )
         if tolerance is not None:
-            indexer = self._filter_indexer_tolerance(target, indexer, tolerance)
+            indexer = self._filter_indexer_tolerance(target_values, indexer, tolerance)
         return indexer
 
     def _filter_indexer_tolerance(
-        self, target: "Index", indexer: np.ndarray, tolerance
+        self,
+        target: Union["Index", np.ndarray, ExtensionArray],
```

Comment:
i dont think we want to change this.  instead can you do the same values-> _values edit on L3072?

Reply:
Ah, good call; should be all set.